tasks/development.ts

```diff
@@ -208,4 +224,165 @@ export function load() {
+      const timelock = timeLockFactory.attach(timelockaddress).connect(timelockProposer) as Timelock;
+
+      const blockBefore = await hre.ethers.provider.getBlock("latest");
+      const timestamp = blockBefore.timestamp;
+
+      const executionStartTimestamp = timestamp + 60 * 60; // now + 1h
+
+      const recorder = new Recorder(
+        new TimelockStrategy({
+          timelock: timelock,
+          executionStartTimestamp: executionStartTimestamp
+        })
+      );
+
+      const recordableFlipper = toRc(flipper, recorder);
+
+      console.log("Flipper state[0] before:", await recordableFlipper.state(0));
+      console.log("Flipper state[1] before:", await recordableFlipper.state(1));
+      console.log("Flipper state[2] before:", await recordableFlipper.state(2));
+
+      await recordableFlipper.record.flip(0);
+      await recordableFlipper.record.flip(1);
+
+      const receipts = await recorder.execute();
+      const { txParamsHash, txHash } = extractTxParamsHashAndTxHashFromSingleTransaction(receipts, "QueuedTransactionsBatch");
+      const { txParamsData } = await extractTimelockQueuedTransactionsBatchParamsDataAndHash(hre, txHash);
+
+      console.log("txHash:", txHash);
+      console.log("txParamsHash:", txParamsHash);
+      console.log("txParamsData:", txParamsData);
+    });
+
+  task("dev:approve-timelock-transaction-by-txParamsHash")
+    .addPositionalParam("timelockaddress")
+    .addPositionalParam("txParamsHash", "Transaction input data hash")
+    .setAction(async ({ timelockaddress, txParamsHash }, hre) => {
+      // this is dev only, normally done by multisig which is supposed to be the owner of the timelock
+
+      const timelockFactory = await hre.ethers.getContractFactory("Timelock");
+      const timelock = timelockFactory.attach(timelockaddress) as Timelock;
+
+      await (await timelock.approveTransactionsBatch(txParamsHash)).wait();
+    });
+
+  task("dev:approve-timelock-transaction-by-txHash")
+    .addPositionalParam("timelockaddress")
+    .addPositionalParam("txHash", "Transaction input hash")
+    .setAction(async ({ timelockaddress, txHash }, hre) => {
+      // this is dev only, normally done by multisig which is supposed to be the owner of the timelock
+
+      const { txParamsHash } = await extractTimelockQueuedTransactionsBatchParamsDataAndHash(hre, txHash);
+      console.log("txParamsHash", txParamsHash);
+
+      const timelockFactory = await hre.ethers.getContractFactory("Timelock");
+      const timelock = timelockFactory.attach(timelockaddress) as Timelock;
+
+      await (await timelock.approveTransactionsBatch(txParamsHash)).wait();
+    });
+
+  task("dev:approve-execute-timelock-transaction")
+    .addPositionalParam("timelockaddress")
+    .addPositionalParam("txRawParamsData")
+    .setAction(async ({ timelockaddress, txRawParamsData }, hre) => {
+      // this is dev only, normally done by multisig which is supposed to be the owner of the timelock
```

Comment:
If it's dev only, can we add an if condition if it's `mainnetFork` and if not throw an error?

Reply:
Nope, it's totally harmless and It's actually useful to be able to use it on RSK (e.g. I did tests on RSK deployed contracts)